### ISSUE TITLE
hyprpm: build plugins with `-j$(nproc)`

### ISSUE THREAD

**Issue by ThatOneCalculator** [hyprwm/Hyprland#5997]
### Description

Right now when installing/updating plugins that are built from source with `hyprpm`, they build with only one core, however, most systems would be able to update a lot faster by taking advantage of all cores when building.

**Comment by vaxerski** [comment]
most plugins are very few files, and using all cores can hang the system up drastically though

**Comment by ThatOneCalculator** [comment]
Maybe half of system cores then?

**Comment by thejch** [comment]
if this were to be changed, it should respect user system environment setting (e.g. MAKEFLAGS or CMAKE_BUILD_PARALLEL_LEVEL) instead of hardcoding a number of threads IMO. hogging all cores or even half the cores is undesirable sometimes, and for other people they might want to use all cores and half would be too few for these people

**Comment by MahouShoujoMivutilde** [comment]
I agree with the comment above, for how small plugins typically are it takes way too long (even comparable to the whole hyprland build in chroot by aurutils, on my system).

> using all cores can hang the system up drastically though

Which is completely nuts that that can still happen on a modern system. 

For anyone who cares about responsiveness under a heavy load I suggest `linux-zen` (`linux-rt` is better, obviously, but has issues) or `ananicy-cpp` with rules to set a high `nice` values for any not realtime-critical cpu-hogging application.

**Comment by vaxerski** [comment]
I use zen and even then when cpu usage is close to 100% the system goes to shit :P

**Comment by ThatOneCalculator** [comment]
Strangely enough, I only noticed that one compiling c++ (xanmod). Compiling rust, c, zig, etc and of course definitely makes my system slow down a bit but not crawl to a halt :P

**Comment by fufexan** [comment]
Plugins can be built with cmake + ninja if you want to use all cores.

**Comment by ThatOneCalculator** [comment]
The point of this issue is that there should be some way to configure it for hyprpm.
